### PR TITLE
Fix converter script

### PR DESCRIPTION
**Pull request by fdplacido**
BEGINRELEASENOTES
- convertMarlinSteeringToGaudi: Fix parameter parse exceptions caused by empty strings in parameter list.
- convertMarlinSteeringToGaudi: Add algorithms from if statements in algList as comments 

ENDRELEASENOTES

### CODE REVIEW

File: scripts/convertMarlinSteeringToGaudi.py [hunk 1]
```diff
@@ -122,15 +132,20 @@ def createFooter(lines, glob):
...
   lines.append("%s.Parameters = [" % proc.replace(".", "_"))
   for para in sorted(params):
     if para not in ["type", "Verbosity"]:
       value = params[para].replace('\n', ' ')
       value = " ".join(value.split())
       value = value.replace(" ", "\", \"")
-      lines.append("%s\"%s\", \"%s\", END_TAG," % (' ' * (len(proc) + 15), para, value))
+
+      if not value:
+          lines.append("%s\"%s\", END_TAG," % (' ' * (len(proc) + 15), para))
```
Comment (andresailer) [inline scripts/convertMarlinSteeringToGaudi.py:145]: too much indentation

File: scripts/convertMarlinSteeringToGaudi.py [hunk 2]
```diff
@@ -13,19 +13,21 @@ def getTree(xmlFile):
   et.parse(xmlFile)
   return et
 
-
```
Comment (andresailer) [inline scripts/convertMarlinSteeringToGaudi.py:16]: keep this empty line (2 empty lines before function definitions)

File: scripts/convertMarlinSteeringToGaudi.py [hunk 3]
```diff
@@ -13,19 +13,21 @@ def getTree(xmlFile):
...
   for proc in procElements:
     procName = proc.attrib.get("name")
     paramDict = dict()
     paramDict["type"] = proc.attrib.get("type")
+
     parameters = proc.findall('parameter')
     for param in parameters:
       paramName = param.attrib.get('name')
-      paramDict[paramName] = getValue(param, "<This Value did not Exist in this file or was empty>")
+      # paramDict[paramName] = getValue(param, "<This Value did not Exist in this file or was empty>")
```
Comment (andresailer) [inline scripts/convertMarlinSteeringToGaudi.py:29]: remove the commented lines.

File: scripts/convertMarlinSteeringToGaudi.py [hunk 4]
```diff
@@ -88,6 +94,10 @@ def getExecutingProcessors(lines, tree):
   execGroup = tree.findall('group')
 
   for proc in execProc:
+    if proc.tag == "if":
+      for child in proc:
+        if child.tag == "processor":
+          lines.append("# algList.append(%s)" % child.get('name').replace(".", "_"))
```
Comment (andresailer) [inline scripts/convertMarlinSteeringToGaudi.py:100]: would it be possible to do 
```suggestion
          lines.append("# algList.append(%s)  # %s" % (child.get('name').replace(".", "_"), <VALUE OF THE IF TAG>))
```
?

Comment (fdplacido) [reply scripts/convertMarlinSteeringToGaudi.py:100]: yes, I'll add it.

**Comment by fdplacido** [comment]
I addressed all your comments.

**Pull request merged**